### DISCUSSION TITLE
Is it safe to pass functions(callbacks) as parameters?

### DISCUSSION THREAD

**Discussion by suhdonghwi** [react-navigation/react-navigation#12121] [Q&A] [answered]
While using the library, I discovered that even when passing a function as a param in `navigation.push` to navigate between routes, it works just fine.

I had assumed that only JSON-serializable values would work. Is there any guarantee that this will always work?

The official documentation mentions following:

> We recommend that the params you pass are JSON-serializable. That way, you'll be able to use [state persistence](https://reactnavigation.org/docs/7.x/state-persistence) and your screen components will have the right contract for implementing [deep linking](https://reactnavigation.org/docs/7.x/deep-linking).

 Regarding this, does that mean it's okay to use this as long as I'm not using features like state persistence or deep linking?

Thank you.

**Comment by satya164** [comment, marked as answer]
Yeah, unless you use these features, the navigation state is never serialized and it's safe. However it's probably better to avoid it regardless to avoid future refactors in case you need to support things like deep linking, web or state persistence, and there are already alternative ways to do the same things.

**Reply by suhdonghwi** [reply]
Thank you. What are alternative ways to do the same things(= cross-screen communication)? I have tried using custom event emitters but it made code readability much worse.

**Reply by satya164** [reply]
Depends on your use case. For passing data to a previous screen, you can pass `params` with [`popTo`](https://reactnavigation.org/docs/7.x/params#passing-params-to-a-previous-screen) on v7 or with [`navigate`](https://reactnavigation.org/docs/params#passing-params-to-a-previous-screen) on v6.
